admin/services/check-complete.service.js

```diff
@@ -25,8 +25,10 @@ checkCompleteService.completeCheck = async function (completedCheck) {
     await pupilDataService.sqlUpdate({ id: pupil.id, pinExpiresAt: moment.utc(), pin: null })
   }
   // Timestamp the request
-  // TODO: This timestamp should be recorded in the application service tier instead
-  completedCheck.receivedByServerAt = moment.utc()
+  const receivedByServerAt = moment.utc()
+
+  // update receivedByServerAt request timestamp
+  await checkDataService.sqlUpdateCheckReceivedByServerAt(completedCheck.data.pupil.checkCode, receivedByServerAt)
```

Comment:
Good work, however, I think the intention here is to timestamp the reception of the data, so wouldn't it make more sense to alter `completedCheckDataService.sqlAddResult()` which is called from `checkCompleteService.completeCheck()`

Reply:
Agreed after discussion - changes made 😃